src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -155,7 +155,9 @@ class ProtobufPlugin implements Plugin<Project> {
      */
     private Object getSourceSets() {
       if (Utils.isAndroidProject(project)) {
-        return project.android.sourceSets
+        // IntelliJ's Android Support plugin adds the 'android' project property even for
+        // non-Android Gradle projects, so fall back on project.sourceSets if necessary
+        return project.android.sourceSets ?: project.sourceSets
```

Comment:
I think it's better to fix `Utils.isAndroidProject()` with the same idea, and leave this method as-is.